File: backend/src/main/java/dev/agiro/fanel/recipes/api/RecipeDto.java
package dev.agiro.fanel.recipes.api;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record RecipeDto(UUID id, UUID householdId, String name, int servings, String notes,
                        String description, List<String> steps, List<String> tags,
                        String imageMimeType, String imageData,
                        List<IngredientDto> ingredients, Instant createdAt) {
}

File: android/app/src/main/java/dev/agiro/fanel/android/data/remote/RecipeDto.kt
package dev.agiro.fanel.android.data.remote

data class IngredientDto(
    val id: String?,
    val name: String,
    val quantity: Double?,
    val unit: String?,
    val category: String?
)

data class RecipeDto(
    val id: String,
    val householdId: String,
    val name: String,
    val servings: Int,
    val notes: String?,
    val description: String?,
    val steps: List<String>?,
    val tags: List<String>?,
    val imageMimeType: String?,
    val imageData: String?,
    val ingredients: List<IngredientDto>?,
    val createdAt: String?
)
